### PR TITLE
Core/BattleGrounds Implement mercenary system

### PR DESCRIPTION
**Pull request by Traesh**
**Changes proposed:**

-  Implement WoD mercenary system
-  Allow player to speak to a NPC in order to gain "Mercenary contract" aura
-  If you queue for BG with this aura, you'll be assigned to the opposite faction team

**Target branch(es):** master

- [x] master

**Issues addressed:** None

**Tests performed:** Built, tested ingame with multiple players

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] Add missing DB data (gossip & smart_script for aura-giver NPC)

### CODE REVIEW

File: src/server/game/Miscellaneous/SharedDefines.h [hunk 1]
```diff
@@ -6231,4 +6231,26 @@ enum class GameError : uint32
...
+    MAP_NORTHREND               = 571,
+    MAP_EBON_HOLD_DK_START_ZONE = 609,
+    MAP_PANDARIA                = 870,
+    MAP_DRAENOR                 = 1116,
+    MAP_WOD_BLASTED_LANDS_PHASE = 1190,
+    MAP_BROKEN_ISLANDS          = 1220,
+    MAP_TANAAN_JUNGLE_INTRO     = 1265,
+    MAP_TANAAN_JUNGLE           = 1464,
+    MAP_DALARAN_UNDERBELLY      = 1502,
+};
```
Comment (soulfrost) [inline src/server/game/Miscellaneous/SharedDefines.h]: whyyyyyyyyyyyyyy

Comment (Traesh) [reply src/server/game/Miscellaneous/SharedDefines.h]: Wups, cherry-picked and didn't remove that enum

**Comment by Traesh** [comment]
@Shauren Any feedback on this one ?

**Comment by Shauren** [comment]
TODO
Aura 193970 still needs implementation - partial info here
https://www.wowhead.com/spell=193475/mercenary-contract#comments:id=2274664 (misses new races and new race/class combinations)
Displayids used arent the default ones for each race

Racial skills

**Pull request merged**